### PR TITLE
Ensure autodiscover events are processed in order

### PR DESCRIPTION
**Pull request by exekias**
We need to read start/stop events from a single queue to ensure they are
processed in order.

Before this change, in some edge cases, when updates spawn stop/start
events, autodiscover was processing the start event before stop, leading
to an undesired state: no running config for an existing pod.

### CODE REVIEW

File: libbeat/autodiscover/autodiscover.go [hunk 1]
```diff
@@ -27,11 +27,8 @@ type Adapter interface {
 	// RunnerFactory provides runner creation by feeding valid configs
 	cfgfile.RunnerFactory
 
-	// StartFilter returns the bus filter to retrieve runner start triggering events
-	StartFilter() []string
-
-	// StopFilter returns the bus filter to retrieve runner stop triggering events
-	StopFilter() []string
+	// EventFilter returns the bus filter to retrieve runner start/stop triggering events
+	EventFilter() []string
```
Comment (ruflin) [inline libbeat/autodiscover/autodiscover.go]: Makes also the interface simpler.

File: libbeat/autodiscover/autodiscover_test.go [hunk 2]
```diff
@@ -173,19 +165,40 @@ func TestAutodiscover(t *testing.T) {
...
+	assert.False(t, runners[1].stopped)
+
+	// Test stop event
+	eventBus.Publish(bus.Event{
+		"stop": true,
+		"meta": common.MapStr{
+			"foo": "baz",
+		},
+	})
+	time.Sleep(10 * time.Millisecond)
```
Comment (ruflin) [inline libbeat/autodiscover/autodiscover_test.go]: Let's hope it's enough on slow machines :-)

**Comment by exekias** [comment]
Thank you for the review! you are right about the moved code blocks, most of it is just wrapping old code into new methods

File: libbeat/autodiscover/autodiscover_test.go (hunk 2 above)
Comment (exekias) [reply libbeat/autodiscover/autodiscover_test.go]: I don't learn the lesson :), let see if jenkins like it, I'll can add some back-off code if not

Comment (ruflin) [reply libbeat/autodiscover/autodiscover_test.go]: It's more a question of how often Jenkins will like it 😆

**Comment by ruflin** [comment]
@exekias I assume the commits can be squashed?

**Comment by exekias** [comment]
Sure

**Comment by exekias** [comment]
Failing test is being addressed here: https://github.com/elastic/beats/pull/6232

**Pull request merged**